src/fixed.rs

```diff
@@ -37,23 +35,20 @@ macro_rules! impl_fixedint {
                 Self::REQUIRED_SPACE
             }
 
-            fn encode_fixed_light<'a>(&'a self) -> &'a [u8] {
+            fn encode_fixed_light(&self) -> &[u8] {
                 return unsafe {
-                    std::slice::from_raw_parts(
-                        transmute::<&$t, *const u8>(&self),
-                        Self::REQUIRED_SPACE,
-                    )
+                    std::slice::from_raw_parts(self as *const $t as *const u8, Self::REQUIRED_SPACE)
                 };
             }
 
             fn encode_fixed(self, dst: &mut [u8]) {
                 assert_eq!(dst.len(), Self::REQUIRED_SPACE);
-                let encoded = unsafe { transmute::<&$t, &[u8; $sz]>(&self) };
+                let encoded = unsafe { &*(&self as *const $t as *const [u8; $sz]) };
```

Comment:
is this to fix a clippy warning or your personal taste? Because I like the old way better :-)

Reply:
This is just to fix clippy warning, I don't have a personal taste for this. We can keep it the old way if it suits you better. :-)